pkg/proxy/proxy.go

```diff
@@ -67,14 +70,77 @@ const (
 	redirectCreationAttempts = 5
 )
 
-// Redirect is the generic proxy redirect interface that each proxy redirect
-// type must export
-type Redirect interface {
-	ToPort() uint16
-	UpdateRules(l4 *policy.L4Filter, wg *completion.WaitGroup) error
-	getSource() ProxySource
+type Redirect struct {
+	mutex lock.RWMutex
+
+	// The following fields are only written to during initialization, it
+	// is safe to read these fields without locking the mutex
+
+	// ProxyPort is the port the redirects redirects to where the proxy is
+	// listening on
+	ProxyPort      uint16
+	endpointID     uint64
+	id             string
+	ingress        bool
+	port           uint16
+	source         ProxySource
+	parserType     policy.L7ParserType
+	created        time.Time
+	implementation RedirectImplementation
+
+	// The following fields are updated while the redirect is life, mutex
+	// must be held to read and write these fields
```

Comment:
Maybe move the mutex here to signify that fields below it need the protection?

Reply:
fixed